sources/packages/backend/libs/integrations/src/services/disbursement-value/disbursement-value.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from "@nestjs/common";
+import { IsNull, MoreThan, Not, Repository } from "typeorm";
+import { DisbursementValue } from "@sims/sims-db";
+import { InjectRepository } from "@nestjs/typeorm";
+
+/**
+ * Service layer for Disbursement value.
+ */
+@Injectable()
+export class DisbursementValueService {
+  constructor(
+    @InjectRepository(DisbursementValue)
+    private readonly disbursementValueRepo: Repository<DisbursementValue>,
+  ) {}
+
+  /**
+   * Checks if there is any partial or full award amount, that
+   * was withheld due to a restriction during an e-cert generation.
+   * @param disbursementScheduleId disbursement schedule id.
+   * @returns true if there is any partial or full award amount, that
+   * was withheld due to a restriction.
+   */
+  async hasAnyFullOrPartialAwardWithheldDueToRestriction(
+    disbursementScheduleId: number,
+  ): Promise<boolean> {
+    return !!(await this.disbursementValueRepo.findOne({
+      select: { id: true },
+      where: {
+        disbursementSchedule: {
+          id: disbursementScheduleId,
+        },
+        restrictionAmountSubtracted: MoreThan(0),
+        restrictionSubtracted: {
```

Comment:
Do we need this criteria ? I thought line 32 could be sufficient.

Reply:
I added it for extra safety.
But thinking about it now, should we need to consider the below scenario for this case
A student has a BCSL award for 5k, and with this 5k they will reach the lifetime maximum (ie, they already had 45k  + this 5k = 50k max lifetime), in that case, there will be a restrictions id but the  substracted amount will be ZERO

Reply:
confirmed with @andrewsignori-aot , Removed restriction id logic